docs/redesign-proposal.md

```diff
@@ -0,0 +1,93 @@
+# Navigator design
+
+Navigator as a system intends to provide a way to easily provision, monitor and
+manage complex systems on top of cluster schedulers, initially targetting
+Kubernetes and OpenShift.
+
+## Architecture
+
+All communication between components of the Navigator system should communicate
+through the `navigator-apiserver`. This allows for a consistent audit trail,
+recorded in a central place. Monitoring services such as `kube-state-metrics`
+can then be used to track changes to the contents of the API server to be
+consumed by something like Prometheus.
```

Comment:
@wallrj 

(1) Regarding the audit trail, this is undecided at present. CustomResourceDefinitions and ThirdPartyResources should be thought of as more of a 'simple database' than anything else. Whilst `kubectl` allows you to interact with them, it does no more other than CRUD operations. If the user is using CRD/TPRs as a storage backend for the navigator-apiserver, then yes there will be CRUD events logged in whatever audit framework the `kube-apiserver` has, however a user may *not* be using CRD/TPR as a storage backend, and regardless Navigator itself should manage this so that it can provide the same functionality between different environments (say, in an environment that doesn't use containers, ie. plain GCE or AWS).

(2) Just like `kube-apiserver`, there is one, although it can scale horizontally (again, like the `kube-apiserver`). The reason they can scale horizontally is that the backing storage (ie. CRDs or TPRs or etcd) is responsible for ensuring consistency.

(3) Access control is initially out of scope. If using the kube-aggregator (in the Kubernetes case) to expose the `navigator-apiserver`, 'delegated authentication' can be used (ie. relying on *another* API server to handle auth). More details/an example is available in the `service-catalog` repo: https://github.com/kubernetes-incubator/service-catalog/blob/master/docs/auth.md

(4) Not just this lack of support for CRDs, but we may at some point be working in an environment completely unlike K8s/OS (ie. targeting a cloud directly perhaps). But yep, worth mentioning.